OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -685,15 +694,36 @@ protected virtual void OnBecomingIdle(Actor self)
 				}
 			}
 
-			var isCircler = !Info.FlightDynamics.HasFlag(FlightDynamic.Hover);
-			if (!atLandAltitude && Info.LandWhenIdle && Info.LandableTerrainTypes.Count > 0)
+			if ((atLandAltitude && !CanLand(self.Location) && ReservedActor == null)
+				|| (!atLandAltitude && altitude != Info.CruiseAltitude && Info.IdleBehavior != IdleBehaviorType.Land))
+			{
+				self.QueueActivity(new TakeOff(self));
+				return;
+			}
+			else if (!atLandAltitude && Info.IdleBehavior == IdleBehaviorType.Land && Info.LandableTerrainTypes.Count > 0)
+			{
 				self.QueueActivity(new Land(self));
-			else if (isCircler && !atLandAltitude)
+				return;
+			}
+			else if (Info.IdleBehavior == IdleBehaviorType.ReturnToBase)
+			{
+				// If we're already on a resupplier, do nothing
```

Comment:
Should we care about `FlightDynamic.TakeOffOnResupply` here?

Reply:
Not sure what you mean...?